server/Controllers/changePassword.js

```diff
@@ -0,0 +1,45 @@
+import bcrypt from "bcrypt";
+import UserModel from "../Schemas/User";
+
+const validatePassword = (password) => {
+  const reg = /(?=^.{8,}$)((?!.*\s)(?=.*[A-Z])(?=.*[a-z]))((?=(.*\d){1,})|(?=(.*\W){1,}))^.*$/;
+  return reg.test(password);
+};
+
+const changePassword = async (req, res) => {
+  const encodedNewPassword = bcrypt.hashSync(req.body.newPassword, 10);
+
+  if (!(await validatePassword(req.body.newPassword))) {
+    res.status(400).send();
+  } else {
+    try {
+      const userInfos = await UserModel.findById(req.userId);
+      if (userInfos === null) {
+        res.status(404).send();
+      } else if (
+        !(await bcrypt.compareSync(req.body.oldPassword, userInfos.password))
+      ) {
+        res.status(401).send();
+      } else {
+        try {
+          const userInfo = await UserModel.findByIdAndUpdate(
+            req.userId,
+            { password: encodedNewPassword },
+            {
+              runValidators: true
+            }
+          );
```

Comment:
I think using `.save()` on `userInfos ` would be better than `findByIdAndUpdate ` (that search a second time).

BUT ! You're gonna merge all that code with `editUser` (`editUserInfo`) so it's gonna change a bit and I let you choose what you prefer